src/cleo/ui/exception_trace/frame_collection.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from typing import List
+
+from cleo.ui.exception_trace.frame import Frame
+
+
+class FrameCollection(List[Frame]):
+    def __init__(self, frames: list[Frame] | None = None, count: int = 0) -> None:
+        if frames is None:
+            frames = []
+
+        super().__init__(frames)
+
+        self._count = count
+
+    @property
+    def repetitions(self) -> int:
+        return self._count - 1
+
+    @property
+    def is_repeated(self) -> bool:
+        return self._count > 1
+
+    def increment_count(self, increment: int = 1) -> FrameCollection:
+        self._count += increment
+
+        return self
+
+    def compact(self) -> list[FrameCollection]:
+        """
+        Compacts the frames to deduplicate recursive calls.
+        """
```

Comment:
Curious: are the PEP 257 violations in the entire repo intended? I don't see any explicit mention in the config etc. 🤔

Reply:
Docs(trings) are notoriously neglected in this project. Hopefully, we will fix that for the 3.0 release.